modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/foreach/ForEachPropertiesTestCase.java

```
@@ -0,0 +1,582 @@
+                    int end = matcher.end();
+                    String quote = payload.substring(start, end);
+
+                    assertTrue(quote.contains(
+                                    "<m0:group>Group2</m0:group>"),
+                            "Group Element not found");
+                    assertTrue(quote.contains(
+                                    "<m0:symbol>Group1_Group2_IBM</m0:symbol>"),
+                            "IBM Element not found");
+                    assertTrue(quote.contains(
+                                    "<m0:symbol>Group1_Group2_WSO2</m0:symbol>"),
+                            "WSO2 Element not found");
+                    assertTrue(quote.contains(
+                                    "<m0:symbol>Group1_Group2_MSFT</m0:symbol>"),
+                            "MSTF Element not found");
+                    assertTrue(quote.contains(
+                                    "<m0:symbol>Group1_Group2_SUN</m0:symbol>"),
+                            "SUN Element not found");
+                }
+            }
+        }
+    }
+
+    @AfterClass
+    public void close() throws Exception {
+        super.cleanup();
+    }
+
+    private void sendRequest(String addUrl, String query)
+            throws IOException {
+        String charset = "UTF-8";
+        URLConnection connection = new URL(addUrl).openConnection();
+        connection.setDoOutput(true);
+        connection.setRequestProperty("Accept-Charset", charset);
+        connection.setRequestProperty("Content-Type",
+                "application/xml;charset=" + charset);
+        OutputStream output = null;
+        try {
+            output = connection.getOutputStream();
+            output.write(query.getBytes(charset));
+        } finally {
+            if (output != null) {
+                try {
+                    output.close();
+                } catch (IOException logOrIgnore) {
+                    log.error("Error while closing the connection");
```

Dont swallow exceptions and it is better to throw exceptions rather catching in test cases